### PR TITLE
front: fix a crash that would happen when editing a train

### LINKED ISSUE #8377: Front crash when edit input table for a specific scenario
### What happened?

I get this error when edit the table : 

https://github.com/user-attachments/assets/46ed2768-db28-4e5d-b952-c0545d881bda

### What did you expect to happen?

The front don't crash

### How can we reproduce it (as minimally and precisely as possible)?

1. Import the [train_schedules.json](https://github.com/user-attachments/files/16549370/train_schedules.json)
2. Edit a train
3. Set any value in Times and stops table 
4. Save the modification 

### On which environments the bug occurs?

Recette (SNCF)

### On which browser the bug occurs?

Google Chrome

### OSRD version (top right corner `Account` button > `Informations`)

02781ca

### PR DESCRIPTION
**Pull request by anisometropie**
Closes #8377.

### CODE REVIEW

**Review by Uriel-Sautron — APPROVED** [review]
LGTM!! Tested

**Review by SarahBellaha — APPROVED** [review]
LGTM :)

File: front/src/applications/operationalStudies/hooks/useLazyLoadTrains.ts [hunk 1]
```diff
@@ -82,19 +82,33 @@ const useLazyLoadTrains = ({
...
+
+        if (!outOfSync) {
+          // launch the projection of the trains
+          setTrainIdsToProject((prev) => [...prev, ...packageToFetch]);
+
+          // format the summaries to display them in the timetable
+          const newFormattedSummaries = formatTrainScheduleSummaries(
+            packageToFetch,
+            rawSummaries, // rawSummaries
+            trainSchedulesById, // rawTrainSchedules Map<number, TrainScheduleResult>
```
Comment (clarani) [inline front/src/applications/operationalStudies/hooks/useLazyLoadTrains.ts]:
```suggestion
            rawSummaries,
            trainSchedulesById,
```

**Review by clarani — APPROVED** [review]
LGTM, tested :white_check_mark:

File: front/src/applications/operationalStudies/hooks/useLazyLoadTrains.ts (hunk 1 above)
Comment (anisometropie) [reply front/src/applications/operationalStudies/hooks/useLazyLoadTrains.ts]: done

**Pull request merged**